frontend/src/pages/pipelines/global/experiments/ArchiveExperimentModal.tsx

```diff
@@ -0,0 +1,44 @@
+import { StackItem } from '@patternfly/react-core';
+import React from 'react';
+import ArchiveModal from '~/concepts/pipelines/content/ArchiveModal';
+import { usePipelinesAPI } from '~/concepts/pipelines/context';
+import { ExperimentKFv2 } from '~/concepts/pipelines/kfTypes';
+
+interface ArchiveExperimentModalProps {
+  experiment: ExperimentKFv2;
+  onCancel: () => void;
+  isOpen: boolean;
+}
+
+export const ArchiveExperimentModal: React.FC<ArchiveExperimentModalProps> = ({
```

Comment:
Now when it comes to if you should combine the experiment modals, I personally would combine the bulk and the single experiment modals so 2 components total giving you ArchiveExperiments and RestoreExperiments.

This is a bit of a nit, but i think it would clean up a lot of the logic around choosing between one of the other.